src/main/java/brave/opentracing/BraveActiveSpanSource.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.opentracing;
+
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Manages the active Brave span.
+ *
+ * Note that it is important to use the OpenTracing API exclusively to activate spans. If you fail
+ * to do this, querying for the active span using the OpenTracing API will throw an exception.
+ */
+final class BraveActiveSpanSource implements ActiveSpanSource {
+  private final Map<Long, BraveActiveSpan> activeSpans = new ConcurrentHashMap<>();
```

Comment:
If I use my own ThreadLocal to track the active span, I wouldn't be able to stay in sync if an instrumenter used the OT API to make a span active, but then finished it with the Brave API. (That's actually how I tackled this in the first place, where my original impl used ot-util's ThreadLocalActiveSpanSource.)

The underlying thing here is that I need to depend on Brave to tell me what the active span is, but Brave doesn't offer the ability to access or close the scope associated with the current span unless you have access to the object that Brave returned when you made the span active in the first place.